mlflow/lightgbm.py

```diff
@@ -0,0 +1,175 @@
+"""
+The ``mlflow.lightgbm`` module provides an API for logging and loading LightGBM models.
+This module exports LightGBM models with the following flavors:
+
+LightGBM (native) format
+    This is the main flavor that can be loaded back into LightGBM.
+:py:mod:`mlflow.pyfunc`
+    Produced for use by generic pyfunc-based deployment tools and batch inference.
+"""
+
+from __future__ import absolute_import
+
+import os
+import yaml
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.models import Model
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.exceptions import MlflowException
+
+FLAVOR_NAME = "lightgbm"
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`.
+    """
+    import lightgbm as lgb
+
+    return _mlflow_conda_env(
+        additional_conda_deps=None,
+        # LightGBM is not yet available via the default conda channels, so we install it via pip
+        additional_pip_deps=[
+            "lightgbm=={}".format(lgb.__version__),
+        ],
+        additional_conda_channels=None)
+
+
+def save_model(lgb_model, path, conda_env=None, mlflow_model=Model()):
+    """
+    Save a LightGBM model to a path on the local file system.
+
+    :param lgb_model: LightGBM model to be saved.
```

Comment:
```suggestion
    :param lgb_model: LightGBM model (an instance of lightgbm.Booster) to be saved.
```

Assuming my understanding is correct that the `save_model`, `log_model`, `load_model` APIs only work for `lightgbm.Booster` models, and not models that implement the [scikit-learn API](https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api) - it'd also be good to add this clarification to the docstrings of `log_model` and `load_model`

Reply:
@smurching Thanks for the comment. I'll add this clarification to XGBoost flavor as well.

Reply:
@smurching Do you think we should warn users that LightGBM and XGBoost flavor don't yet support models that implement the scikit-learn API?

Reply:
Yeah I think that'd definitely be helpful, I'd be happy to review a docstring update that adds such a warning, for example:

```suggestion
    :param lgb_model: LightGBM model (instance of `lightgbm.Booster <https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.Booster.html#lightgbm.Booster>`_) to be saved. Note that saving LightGBM models that implement the `scikit-learn API <https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api>`_  is not supported.
```

Reply:
@smurching Thanks. I'll open a PR to add this warning.

Reply:
@smurching I opened a PR to update docstring https://github.com/mlflow/mlflow/pull/2170